File: 0001-0500/0463.Island Perimeter.cpp
class Solution {
public:
    int islandPerimeter(vector<vector<int>>& grid) {
        int sideCnt = 0;
        if(grid.size()!=0 && grid[0].size()!=0){
            for(int i=0;i<grid.size();i++){
                sideCnt += grid[i][0];
                for(int j=0;j<grid[0].size()-1;j++){
                    sideCnt += grid[i][j]^grid[i][j+1];
                }
                sideCnt += grid[i][grid[i].size()-1];
            }
            for(int i=0;i<grid.size()-1;i++){
                for(int j=0;j<grid[0].size();j++){
                    sideCnt += grid[i][j]^grid[i+1][j];
                }
            }
            for(int j=0;j<grid[0].size();j++){
                sideCnt+=grid[0][j]+grid[grid.size()-1][j];
            }
        }
        return sideCnt;
    }
};

File: 0001-0500/0463.Island Perimeter.py
class Solution(object):
    def islandPerimeter(self, grid):
        """
        :type grid: List[List[int]]
        :rtype: int
        """
        DELTA = (1, 0, -1, 0, 1)
        def isIsland(i, j):
            return 0 <= i < len(grid) and 0 <= j < len(grid[0]) and bool(grid[i][j])
        def dfs(i, j):
            if isIsland(i, j):
                if grid[i][j] == 2:
                    return 0
                grid[i][j] = 2
                res = 0
                for k in xrange(4):
                    res += dfs(i + DELTA[k], j + DELTA[k + 1])
                return res
            else:
                return 1
        for i in xrange(len(grid)):
            for j in xrange(len(grid[0])):
                if grid[i][j]:
                    return dfs(i, j)
        return 0
